rest/blip_sync.go

```diff
@@ -51,17 +51,22 @@ func (h *handler) handleBLIPSync() error {
 	// error is checked at the time of database load, and ignored at this time
 	originPatterns, _ := hostOnlyCORS(h.db.CORS.Origin)
 
+	cancelCtx, cancelCtxFunc := context.WithCancel(h.db.DatabaseContext.CancelContext)
 	// Create a BLIP context:
-	blipContext, err := db.NewSGBlipContext(h.ctx(), "", originPatterns, h.db.DatabaseContext.CancelContext)
+	blipContext, err := db.NewSGBlipContext(h.ctx(), "", originPatterns, cancelCtx)
 	if err != nil {
+		cancelCtxFunc()
```

Comment:
Is this the only error exit where we need to ensure cancelCtxFunc() is called?  Is it needed in the PostHandshakeCallback below, or is that handled elsewhere?

Reply:
This is handled in `BlipSyncContext.Close`